frontend/components/NewLayout/Courses/CourseGrid.tsx

```diff
@@ -317,81 +286,38 @@ function CourseGrid() {
-              {moduleTags.map((tag) => (
+            {Object.keys(tags).map((category) => (
+              <div key={category}>
+                <div style={{ gridArea: `${category}Tags` }}>
+                  {tags[category].map((tag) => (
+                    <TagButton
+                      id={`tag-${category}-${tag.id}`}
+                      key={`tag-${category}-${tag.id}`}
+                      variant={
+                        activeTags.includes(tag) ? "contained" : "outlined"
+                      }
+                      onClick={() => handleClick(tag)}
+                      size="small"
+                    >
+                      {tag.name}
+                    </TagButton>
+                  ))}
+                </div>
                 <TagButton
-                  id={`module-tag-${tag}`}
-                  key={`module-tag-${tag}`}
-                  variant={activeTags.includes(tag) ? "contained" : "outlined"}
-                  onClick={() => handleClick(tag)}
+                  id={`select-all-${category}-tags`}
+                  variant={
+                    tags[category].every((tag) => activeTags.includes(tag))
+                      ? "contained"
+                      : "outlined"
+                  }
+                  style={{ gridArea: `${category}SelectAll` }}
+                  onClick={() => handleSelectAllClick(category)}
                   size="small"
                 >
-                  {tag}
+                  {t("selectAll")}
                 </TagButton>
-              ))}
-            </ModuleTagsContainer>
-            <SelectAllModulesButton
-              id="select-all-module-tags"
-              variant={
-                moduleTags.every((tag) => activeTags.includes(tag))
-                  ? "contained"
-                  : "outlined"
-              }
-              onClick={() => handleSelectAllClick("module")}
-              size="small"
-            >
-              {t("selectAll")}
-            </SelectAllModulesButton>
-            <LanguageTagsContainer>
-              {languageTags.map((tag) => (
-                <TagButton
-                  id={`language-tag-${tag}`}
-                  key={`language-tag-${tag}`}
-                  variant={activeTags.includes(tag) ? "contained" : "outlined"}
-                  onClick={() => handleClick(tag)}
-                  size="small"
-                >
-                  {tag}
-                </TagButton>
-              ))}
-            </LanguageTagsContainer>
-            <SelectAllLanguagesButton
-              id="select-all-language-tags"
-              variant={
-                languageTags.every((tag) => activeTags.includes(tag))
-                  ? "contained"
-                  : "outlined"
-              }
-              onClick={() => handleSelectAllClick("language")}
-              size="small"
-            >
-              {t("selectAll")}
-            </SelectAllLanguagesButton>
+              </div>
+            ))}
```

Comment:
This moved the `Select all` -filter buttons to the same grid areas as the individual tag filters inside `TagsContainer`